src/proficiency.cpp

```diff
@@ -31,11 +32,58 @@ bool proficiency_id::is_valid() const
     return proficiency_factory.is_valid( *this );
 }
 
+namespace io
+{
+template<>
+std::string enum_to_string<proficiency_bonus_type>( const proficiency_bonus_type data )
+{
+    // *INDENT-OFF*
+    switch( data ) {
+        case proficiency_bonus_type::strength:
+            return "strength";
+        case proficiency_bonus_type::dexterity:
+            return "dexterity";
+        case proficiency_bonus_type::intelligence:
+            return "intelligence";
+        case proficiency_bonus_type::perception:
+            return "perception";
+        case proficiency_bonus_type::last:
+            break;
+    }
+    // *INDENT-ON*
+
+    debugmsg( "Invalid proficiency bonus type" );
+    return "";
+}
+} // namespace io
+
 void proficiency::load_proficiencies( const JsonObject &jo, const std::string &src )
 {
     proficiency_factory.load( jo, src );
 }
 
+void proficiency::load_proficiency_bonus( const JsonObject &jo )
+{
+    std::string member = "bonuses";
+    if( jo.has_member( member ) ) {
+        if( jo.has_array( member ) ) {
+            for( const JsonValue entry : jo.get_array( member ) ) {
+                if( entry.test_object() ) {
+                    JsonObject obj = entry.get_object();
+
+                    const std::string category = obj.get_string( "category", "" );
+                    if( category != "" ) {
+                        std::string type_str = obj.get_string( "type", "last" );
+                        proficiency_bonus_type prof_bonus_type = io::string_to_enum<proficiency_bonus_type>( type_str );
+                        int value = obj.get_int( "value", 0 );
+                        _bonuses[category].push_back( proficiency_bonus( prof_bonus_type, value ) );
+                    }
+                }
+            }
+        }
+    }
+}
+
```

Comment:
You should just be able to add a deserialize function `proficiency_bonus`, and skip all the direct JSON manipulation, just have optional do it for you below (that's how you get the format I put forth in my first review).

```suggestion
void proficiency_bonus::deserialize( JsonIn &jsin )
{
    const JsonObject &jo = jsin.get_object()
    
    mandatory( jo, false, "type", type );
    mandatory( jo, false, "value", value );
}

```